### PR TITLE
fix(@aws-amplify/auth): incorrect return type for Auth.resendSignUp

### PR DESCRIPTION
**Pull request by mousedownmike**
Partial fix for #5029 

Updated `Auth.resendSignUp` signature to return `any` type instead of `string`.  Revised Spy to be reflective of an actual `CodeDeliveryDetails` response.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -388,12 +388,12 @@ export default class AuthClass {
...
 	 * @param {String} username - The username to be confirmed
 	 * @param {ClientMetadata} clientMetadata - Metadata to be passed to Cognito Lambda triggers
-	 * @return - A promise resolves data if success
+	 * @return - A promise resolves code delivery details if successful
 	 */
 	public resendSignUp(
 		username: string,
 		clientMetadata: ClientMetaData = this._config.clientMetadata
-	): Promise<string> {
+	): Promise<any> {
```
Comment (ericclemmons) [inline packages/auth/src/Auth.ts]: It possible to re-use `CodeDeliveryDetails` here?

https://github.com/aws-amplify/amplify-js/blob/ac34816df326331cfe04474fdf35790c52f4a1d3/packages/amazon-cognito-identity-js/index.d.ts#L6-L10

Comment (mousedownmike) [reply packages/auth/src/Auth.ts]: @ericclemmons are we sure it always resolves to `CodeDeliveryDetails`?  I wasn't positive so I chose `any`.  The code builds fine with `CodeDelieryDetails` in there and I can push that if you'd like.

Comment (mousedownmike) [reply packages/auth/src/Auth.ts]: The branch with `CodeDeliveryDetails` is here: 
https://github.com/mousedownmike/amplify-js/tree/resendsignup_codedeliverydetails

Happy to close this PR and start a new one or merge the change into this one.  Whichever solution works best and whenever it works.  

Cheers,
Mike

**Comment by ericclemmons** [comment]
Everything looks good, but we'll need to manually validate the response payload against the types to be sure whether `any` is the correct type.

The underlying call indicates a `string`, but we have to be certain:
https://github.com/aws-amplify/amplify-js/blob/ac34816df326331cfe04474fdf35790c52f4a1d3/packages/amazon-cognito-identity-js/src/CognitoUser.js#L1280-L1286

**Comment by f22hd** [comment]
Exactly what we added to our project to resolve this mistype of the return type.
Hope this PR approved and closed. Thanks.

File: packages/auth/src/Auth.ts (hunk 1 above)
Comment (ericclemmons) [reply packages/auth/src/Auth.ts:387]: I'm going to get this merged in once tests pass.

We'll do some more tests to confirm the payload shape, but for now this is a great step forward 👍

Comment (ericclemmons) [reply packages/auth/src/Auth.ts:387]: (I went ahead & opened https://github.com/aws-amplify/amplify-js/pull/6561 for comparison)

**Pull request merged**